documentation/hpcc-tf-for-developers.md

```diff
@@ -0,0 +1,373 @@
+| `locals.tf` | Contains local variables used in the deployment of the `storage`. Variables in this file also use some of the easy deploy variables in lite-variables.tf. |
+| `main.tf` | Contains only the `storage` module |
+| `outputs.tf` | Contains only the resource `local_file` which outputs to a file config.json. This is done only when on a successful deployment of `storage`. |
+| `providers.tf` | Contains only 2 providers: azurerm and azuread |
+| `storage.auto.tfvars` | Contains variables that describe the storage accounts that are created. This file is copied to the `storage` directory by the `deploy` script. |
+| `variables.tf` | Contains variables needed for `storage` deployment. |
+| `versions.tf` | dummy description text |
+| <font color="blue">`data`</font> | This directory contains the file `config.json` which is created when the external storage successfully deploys. |
+
+## vnet subdirectory
+
+
+The following table tells what files and subdirectories are in the `vnet` subdirectory. The deployment of an `vnet` happens in this directory. If one deploys from the root directory, the `deploy` script goes to this directory to deploy `vnet`. Also, if you deploy `vnet` manually you do it from this directory.
+
+What is deployed by this subdirectory and their order is given in  [_Appendix D_](#Appendix-D).
+
+|vnet subdirectory entry name|description|
+|:--------------------------------|:----------|
+| `data.tf` | Contains `data` statements providing information about existing resources. |
+| `lite-variables.tf` | Contains all variables used for easy deployment. This file is copied in the hpcc directory by the `deploy` script. |
+| `lite.auto.tfvars` | Contains all the variables used for easy deployment with values of the user. This file is copied in the `vnet` directory by the `deploy` script. |
+| `locals.tf` | Contains local variables used in the deployment of `vnet`. Variables in this file also use some of the easy deploy variables in lite-variables.tf. |
+| `main.tf` | Contains modules and resources needed for the deployment of `vnet` |
+| `outputs.tf` | Contains several output statements that output important information to the deployer. Also, this file contains an output state that outputs a file, config.json. This file is only output if there is a successful deployment of `vnet`. |
+| `providers.tf` |  Contains 2 providers: `random` and `azurerm`. |
+| `variables.tf` | Contains only one variable, `disable_naming_conventions`. |
+| `versions.tf` | Contains the required versions of `terraform`, `azurerm` and `random`. |
+| `vnet.tf` | Contains the module `virtual_network` which deploys the virtual network used by `aks`, `hpcc`, and `storage`. |
+| <font color="blue">`data`</font> | This directory contains the file `config.json` which is created when the `vnet` is successfully deploys. |
+
+​     
+
+## Appendix A 
```

Comment:
None of the information in these appendices adds value.  Please delete all of them.

Reply:
fixed